enhancements/proposals/cleanup_after_scaling.md

```diff
@@ -0,0 +1,140 @@
+# Run node cleanup after scaling the cluster
+
+## Summary
+
+The Operator supports both horizontal and vertical scaling, but the procedure isn’t in sync with ScyllaDB documentation,
+because the cleanup part is not implemented. It’s important because upon scaling, the stored on node disk might become
+stale taking up unnecessary space. Operator should support running the cleanup to keep disk space as low as possible and
+ensure clusters are stable and reliable over time.
+
+## Motivation
+
+The motivation behind running node cleanup after scaling a cluster is to avoid the accumulation of unnecessary data on
+the node disks. When nodes are added or removed from the cluster, they gain or lose some tokens, which can result in
+files stored on the node disks still containing data associated with lost tokens. Over time, this can lead to a build-up
+of unnecessary data and cause disk space issues. By running node cleanup after scaling, these files can be cleared,
+freeing up disk space.
+
+### Goals
+
+* Running a node cleanup procedure after horizontally scaling the cluster.
+* Running a node cleanup procedure after scaling is completed, on all nodes but the last one added.
+
+### Non-Goals
+
+Running node cleanup during off-peak hours.
```

Comment:
We can add: Running node cleanup after vertical scaling.
(Since it is not needed in this case)

Reply:
done